tests/unit/src/test/scala/tests/RenameFilesLspSuite.scala

```diff
@@ -0,0 +1,332 @@
+        |package B
+        |import <<A>>.{B => BB}
+        |import <<A>>._ 
+        |object Moon {
+        |  println(BB.Sun)
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A" -> s"$prefix/C"),
+    expectedRenames = Map("A" -> "C"),
+    sourcesAreCompiled = true,
+  )
+
+  renamed(
+    "references-dir-rename-no-new-imports",
+    s"""|/$prefix/A/B/Sun.scala
+        |package <<A>>.B
+        |object Sun
+        |/$prefix/A/B/Moon.scala
+        |package <<A>>.B
+        |object Moon {
+        |  println(Sun)
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A" -> s"$prefix/C"),
+    expectedRenames = Map("A" -> "C"),
+    sourcesAreCompiled = true,
+  )
+
+  renamed(
+    "references-multi-package-new-imports",
+    s"""|/$prefix/A/B/Sun.scala
+        |package A.<<B>>
+        |object Sun
+        |/$prefix/A/B/Moon.scala
+        |package A
+        |package B
+        |<<//>>
+        |object Moon {
+        |  println(Sun)
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A/B/Sun.scala" -> s"$prefix/A/B/C/Sun.scala"),
+    expectedRenames = Map("B" -> "B.C", "//" -> "\nimport A.B.C.Sun//"),
+    sourcesAreCompiled = true,
+  )
+
+  renamed(
+    "references-mixed-import",
+    s"""|/$prefix/A/Mars.scala
+        |package <<A>>
+        |object Phobos
+        |object Deimos
+        |/$prefix/A/Pluto.scala
+        |package A
+        |object Pluto
+        |/$prefix/B/Mars.scala
+        |package B
+        |import A.{<<Phobos => P, >>Pluto<<, Deimos>>}
+        |<<//>>
+        |object Mars
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A/Mars.scala" -> s"$prefix/C/Mars.scala"),
+    expectedRenames = Map(
+      "A" -> "C",
+      "Phobos => P, " -> "",
+      ", Deimos" -> "",
+      "//" -> "import C.{Phobos => P, Deimos}//",
+    ),
+    sourcesAreCompiled = true,
+  )
+
+  /* Cases that are not yet supported */
+  renamed(
+    "unsupported-qualified-refs",
+    s"""|/$prefix/A/B/Sun.scala
+        |package <<A>>.B
+        |object Sun 
+        |/$prefix/X/Moon.scala
+        |package X
+        |<<//>>should not import anything
+        |object Moon {
+        |  A.B.Sun // should be refactored
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A" -> s"$prefix/C"),
+    expectedRenames = Map("A" -> "C", "//" -> "\nimport C.B.Sun//"),
+    sourcesAreCompiled = true,
+  )
+
+  renamed(
+    "unsupported-package-renames",
+    s"""|/$prefix/A/B/Sun.scala
+        |package A.<<B>>
+        |object Sun 
+        |/$prefix/X/Moon.scala
+        |package X
+        |<<//>>should not import anything
+        |import A.{B => BB} // should be refactored
+        |object Moon {
+        |  println(BB.Sun)
+        |}
+        |""".stripMargin,
+    fileRenames = Map(s"$prefix/A/B" -> s"$prefix/A/C"),
+    expectedRenames = Map("B" -> "C", "//" -> "\nimport A.C.Sun//"),
+    sourcesAreCompiled = true,
+  )
```

Comment:
I tested the PR locally and found one minor issue with position of import:
```
a/src/main/scala/one
package one

import scala.concurrent.Future

class A(b: B)
a/src/main/scala/one
class B

// move B to `a/src/main/scala/one/two`

a/src/main/scala/one
package one

import one.two.B // <- it would better to place it  as the last import as we do for auto-imports
import scala.concurrent.Future

class A(b: B)
a/src/main/scala/one/two
class B
```

Reply:
Fixed, here is a [test](https://github.com/susliko/metals/blob/feature-45/tests/unit/src/test/scala/tests/RenameFilesLspSuite.scala#L255)